.github/workflows/container-image.yml

```diff
@@ -0,0 +1,71 @@
+name: container-image-build
+
+on:
+  workflow_call:
+    inputs:
+      experimental:
+        type: boolean
+
+env:
+  NPROC: 2
+  MAKEFLAGS: "-j${NPROC}"
+  NIMFLAGS: "--parallelBuild:${NPROC}"
+
+jobs:
+  build-docker-image:
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
+
+    name: docker-build-${{ matrix.os }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Get submodules hash
+        id: submodules
+        run: |
+          echo "hash=$(git submodule status | awk '{print $1}' | sort | shasum -a 256 | sed 's/[ -]*//g')" >> $GITHUB_OUTPUT
+
+      - name: Cache submodules
+        uses: actions/cache@v3
+        with:
+          path: |
+            vendor/
+            .git/modules
+          key: ${{ runner.os }}-vendor-modules-${{ steps.submodules.outputs.hash }}
+
+      - name: Build binaries
+        id: build
+        run: |
+          EXPERIMENTAL=${{ inputs.experimental }}
+
+          make -j${NPROC} V=1 QUICK_AND_DIRTY_COMPILER=1 NIMFLAGS="-d:disableMarchNative" wakunode2 EXPERIMENTAL=${{EXPERIMENTAL}}
```

Comment:
curious if we can have this workflow depend on the [build job](https://github.com/waku-org/nwaku/blob/b785b6ba1588c808e0c5d78c2c8aa4c7cc5794b7/.github/workflows/ci.yml#L57-L76) in the `ci` workflow

we can probably skip the docker image building if the `ci` workflow fails, right?

Reply:
We could, my thinking initially was to provide the image ASAP rather than wait for the CI to finish, but we definitely can change that